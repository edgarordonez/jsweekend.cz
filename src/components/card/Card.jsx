import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const Wrapper = styled.div`
  .card {
    width: 100%;
    display: flex;
    justify-content: flex-start;
    align-items: center;
    padding: 2em 0;

    .airline-group {
      display: flex;
      flex-direction: column;
      justify-content: center;
      align-items: flex-start;

      .airline {
        display: flex;
        justify-content: center;
        align-items: center;
        margin: .7rem;
  
        img {
          margin: .7rem;
        }
      }
    }

    .flight-detail {
      width: 100%;
      display: flex;
      flex-direction: column;
      justify-content: center;
      align-items: center;
      margin: .7rem;
      small {
        display: block;
      }
    }
  }
`;

export const Card = ({ node }) => (
  <Wrapper>
    <div key={node.id} className="card mui--divider-bottom mui--z3">
      <div className="airline-group">
        {
          node.airlines.map((airline, index) => (
            <div key={`airline-${node.id}-${index}`} className="airline">
              <img src={airline.logoUrl} alt={airline.name} />
              <small>{airline.name}</small>
            </div>
          ))
        }
      </div>
      <div className="flight-detail">
        <p>
          {node.departure.airport.city.name} &#8594; {node.arrival.airport.city.name}
          <small>{new Date(node.departure.localTime).toString().split("GMT")[0]}</small>
        </p>
        <h2>
          {node.price.amount.toLocaleString('es-ES', {style: 'currency', currency: 'EUR'})}
        </h2>
      </div>
    </div>
  </Wrapper>
);

Card.propTypes = {
  node: PropTypes.object,
};

export default Card;
