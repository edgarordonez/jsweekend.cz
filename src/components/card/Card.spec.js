import React from 'react';
import { shallow } from 'enzyme';
import { Card } from './Card';

it('renders without crashing', () => {
  const data = {
    node: {
      id: 1,
      airlines: [
        {
          id: 1,
          name: 'Card test flight',
          logoUrl: ''
        }
      ],
      departure: {
        localTime: 'Thu Feb 22 2018 16:26:58 GMT+0100 (CET)',
        airport: {
          city: {
            name: 'BCN'
          }
        }
      },
      arrival: {
        airport: {
          city: {
            name: 'BRNO'
          }
        }
      },
      price: {
        amount: 0
      }
    }
  }
  const card = shallow(<Card node={data.node} />);
  expect(card).toMatchSnapshot();
});
