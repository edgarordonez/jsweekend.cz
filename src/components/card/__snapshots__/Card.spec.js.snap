// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    node={
        Object {
            "airlines": Array [
              Object {
                "id": 1,
                "logoUrl": "",
                "name": "Card test flight",
              },
            ],
            "arrival": Object {
              "airport": Object {
                "city": Object {
                  "name": "BRNO",
                },
              },
            },
            "departure": Object {
              "airport": Object {
                "city": Object {
                  "name": "BCN",
                },
              },
              "localTime": "Thu Feb 22 2018 16:26:58 GMT+0100 (CET)",
            },
            "id": 1,
            "price": Object {
              "amount": 0,
            },
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <div
        className="card mui--divider-bottom mui--z3"
>
        <div
                className="airline-group"
        >
                <div
                        className="airline"
                >
                        <img
                                alt="Card test flight"
                                src=""
                        />
                        <small>
                                Card test flight
                        </small>
                </div>
        </div>
        <div
                className="flight-detail"
        >
                <p>
                        BCN
                         → 
                        BRNO
                        <small>
                                Thu Feb 22 2018 16:26:58 
                        </small>
                </p>
                <h2>
                        0
                         €
                </h2>
        </div>
</div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": "1",
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="airline-group"
>
            <div
                        className="airline"
            >
                        <img
                                    alt="Card test flight"
                                    src=""
                        />
                        <small>
                                    Card test flight
                        </small>
            </div>
</div>,
          <div
            className="flight-detail"
>
            <p>
                        BCN
                         → 
                        BRNO
                        <small>
                                    Thu Feb 22 2018 16:26:58 
                        </small>
            </p>
            <h2>
                        0
                         €
            </h2>
</div>,
        ],
        "className": "card mui--divider-bottom mui--z3",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="airline"
>
                <img
                                alt="Card test flight"
                                src=""
                />
                <small>
                                Card test flight
                </small>
</div>,
            ],
            "className": "airline-group",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "airline-1-0",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <img
                    alt="Card test flight"
                    src=""
/>,
                  <small>
                    Card test flight
</small>,
                ],
                "className": "airline",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "alt": "Card test flight",
                    "src": "",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "img",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Card test flight",
                  },
                  "ref": null,
                  "rendered": "Card test flight",
                  "type": "small",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p>
                BCN
                 → 
                BRNO
                <small>
                                Thu Feb 22 2018 16:26:58 
                </small>
</p>,
              <h2>
                0
                 €
</h2>,
            ],
            "className": "flight-detail",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "BCN",
                  " → ",
                  "BRNO",
                  <small>
                    Thu Feb 22 2018 16:26:58 
</small>,
                ],
              },
              "ref": null,
              "rendered": Array [
                "BCN",
                " → ",
                "BRNO",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Thu Feb 22 2018 16:26:58 ",
                  },
                  "ref": null,
                  "rendered": "Thu Feb 22 2018 16:26:58 ",
                  "type": "small",
                },
              ],
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  0,
                  " €",
                ],
              },
              "ref": null,
              "rendered": Array [
                0,
                " €",
              ],
              "type": "h2",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <div
          className="card mui--divider-bottom mui--z3"
>
          <div
                    className="airline-group"
          >
                    <div
                              className="airline"
                    >
                              <img
                                        alt="Card test flight"
                                        src=""
                              />
                              <small>
                                        Card test flight
                              </small>
                    </div>
          </div>
          <div
                    className="flight-detail"
          >
                    <p>
                              BCN
                               → 
                              BRNO
                              <small>
                                        Thu Feb 22 2018 16:26:58 
                              </small>
                    </p>
                    <h2>
                              0
                               €
                    </h2>
          </div>
</div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": "1",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="airline-group"
>
              <div
                            className="airline"
              >
                            <img
                                          alt="Card test flight"
                                          src=""
                            />
                            <small>
                                          Card test flight
                            </small>
              </div>
</div>,
            <div
              className="flight-detail"
>
              <p>
                            BCN
                             → 
                            BRNO
                            <small>
                                          Thu Feb 22 2018 16:26:58 
                            </small>
              </p>
              <h2>
                            0
                             €
              </h2>
</div>,
          ],
          "className": "card mui--divider-bottom mui--z3",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="airline"
>
                  <img
                                    alt="Card test flight"
                                    src=""
                  />
                  <small>
                                    Card test flight
                  </small>
</div>,
              ],
              "className": "airline-group",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "airline-1-0",
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <img
                      alt="Card test flight"
                      src=""
/>,
                    <small>
                      Card test flight
</small>,
                  ],
                  "className": "airline",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "alt": "Card test flight",
                      "src": "",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "img",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Card test flight",
                    },
                    "ref": null,
                    "rendered": "Card test flight",
                    "type": "small",
                  },
                ],
                "type": "div",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <p>
                  BCN
                   → 
                  BRNO
                  <small>
                                    Thu Feb 22 2018 16:26:58 
                  </small>
</p>,
                <h2>
                  0
                   €
</h2>,
              ],
              "className": "flight-detail",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "BCN",
                    " → ",
                    "BRNO",
                    <small>
                      Thu Feb 22 2018 16:26:58 
</small>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "BCN",
                  " → ",
                  "BRNO",
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Thu Feb 22 2018 16:26:58 ",
                    },
                    "ref": null,
                    "rendered": "Thu Feb 22 2018 16:26:58 ",
                    "type": "small",
                  },
                ],
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    0,
                    " €",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  0,
                  " €",
                ],
                "type": "h2",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
